strax/sort_enforcement.py

```diff
@@ -0,0 +1,46 @@
+import numpy as np
+from numba.extending import register_jitable
+
+# Define error message as a constant
+UNSTABLE_SORT_MESSAGE = (
+    "quicksort and heapsort are not allowed due to non-deterministic behavior.\n"
+    "Please use mergesort for deterministic sorting behavior."
+)
+
+
+# Define custom exception for sorting errors
+class SortingError(Exception):
+    pass
+
+
+def stable_sort(arr, kind="mergesort", **kwargs):
+    """Stable sort function using mergesort, w/o numba optimization.
+
+    Args:
+        arr: numpy array to sort
+        kind: sorting algorithm to use (only 'mergesort' is allowed)
+
+    Returns:
+        Sorted array using mergesort algorithm
+
+    """
+    if kind != "mergesort":
+        raise SortingError(UNSTABLE_SORT_MESSAGE)
+    return np.sort(arr, kind="mergesort", **kwargs)
+
+
+@register_jitable
```

Comment:
Why `register_jitable` is needed?

Reply:
with `register_jitable`, the function would be called as a regular python function when not used in a `numba` decorated function, and will be compiled as `numba` function if it's called within a decorated function

Reply:
Why not use numba jit as default for `stable_argsort` and `stable_sort `?

Reply:
For `stable_sort` the only supported method is `quicksort`, so we need to avoid it. For `stable_argsort` it's ok to use the `numba` implemented `mergesort`.